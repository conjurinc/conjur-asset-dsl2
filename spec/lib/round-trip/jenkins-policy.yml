- !role &team1
  kind: jenkins-folder
  id: Team1
  owner: !group v4/build

- !role &team2
  kind: jenkins-folder
  id: Team2
  owner: !group v4/build

- &shared_variables
  - !variable &access_key_id aws/target/access_key_id
  - !variable &secret_access_key aws/target/secret_access_key

- !variable &mongo_password_1 mongo/target/team1/db/password
- !variable &mongo_password_2 mongo/target/team2/db/password

- !permit
  resource: *shared_variables
  privilege: execute
  member: *team1

- !permissions
  resource: *mongo_password_1
  privilege: execute
  member: *team1

- !permit
  resource: *shared_variables
  privilege: execute
  member: *team2

- !permissions
  resource: *mongo_password_2
  privilege: execute
  member: *team1
