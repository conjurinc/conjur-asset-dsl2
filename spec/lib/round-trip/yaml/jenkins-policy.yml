- !role &team1
  kind: jenkins-folder
  id: Team1
  owner: !group v4/build

- !role &team2
  kind: jenkins-folder
  id: Team2
  owner: !group v4/build

- &shared_variables
  - !variable &access_key_id aws/target/access_key_id
  - !variable &secret_access_key aws/target/secret_access_key

- !variable &mongo_password_1 mongo/target/team1/db/password
- !variable &mongo_password_2 mongo/target/team2/db/password

- !permit
  role: *team1
  privilege: execute
  resource: *shared_variables

- !permit
  role: *team1
  privilege: execute
  resource: *mongo_password_1
  replace: true

- !permit
  role: *team2
  privilege: execute
  resource: *shared_variables

- !permit
  role: *team1
  privilege: execute
  resource: *mongo_password_2
  replace: true
  