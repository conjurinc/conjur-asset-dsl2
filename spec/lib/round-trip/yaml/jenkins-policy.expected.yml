---
- &3 !role
  id: Team1
  kind: jenkins-folder
  owner: !group
    id: v4/build
- &5 !role
  id: Team2
  kind: jenkins-folder
  owner: !group
    id: v4/build
- - &1 !variable
    id: aws/target/access_key_id
  - &2 !variable
    id: aws/target/secret_access_key
- &4 !variable
  id: mongo/target/team1/db/password
- &6 !variable
  id: mongo/target/team2/db/password
- !permit
  privilege: execute
  resource:
  - *1
  - *2
  role: !member
    role: *3
- !permit
  exclusive: true
  privilege: execute
  resource: *4
  role: !member
    role: *3
- !permit
  privilege: execute
  resource:
  - *1
  - *2
  role: !member
    role: *5
- !permit
  exclusive: true
  privilege: execute
  resource: *6
  role: !member
    role: *3
